app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers.php

```diff
@@ -405,13 +236,17 @@ public function updateSubscriber($subscriber, $updateStatus = false)
                 $subscriber->setData("mailchimp_sync_modified", 0);
                 $saveSubscriber = true;
             } catch (MailChimp_Error $e) {
-                if ($newStatus === 'subscribed' && $subscriber->getIsStatusChanged() && !$helper->isSubscriptionConfirmationEnabled($storeId)) {
+                if ($newStatus === 'subscribed' && $subscriber->getIsStatusChanged()
+                    && !$helper->isSubscriptionConfirmationEnabled($storeId)) {
                     if (strstr($e->getMailchimpDetails(), 'is in a compliance state')) {
                         try {
-                            $api->lists->members->update($listId, $md5HashEmail, null, 'pending', $mergeVars, $interest);
+                            $api->lists->members->update(
```

Comment:
Use the library functions "$api->getLists()->getMembers->update" in stead of calling attributes directly.